R/lasso-development.R

```diff
@@ -165,181 +165,261 @@
       # set to 0 and the other level will be set to 1.
       if (is.factor(private$dfTrainTemp[[self$params$predictedCol]])) {
         private$dfTrainTemp[[self$params$predictedCol]] <-
-          ifelse(private$dfTrainTemp[[self$params$predictedCol]] == levels(private$dfTrainTemp[[self$params$predictedCol]])[1],0,1)
+          ifelse(private$dfTrainTemp[[self$params$predictedCol]] == levels(private$dfTrainTemp[[self$params$predictedCol]])[1],
+                 0,
+                 1)
       }
-
+      
       if (is.factor(private$dfTestTemp[[self$params$predictedCol]])) {
         private$dfTestTemp[[self$params$predictedCol]] <-
-          ifelse(private$dfTestTemp[[self$params$predictedCol]] == levels(private$dfTestTemp[[self$params$predictedCol]])[1],0,1)
+          ifelse(private$dfTestTemp[[self$params$predictedCol]] == levels(private$dfTestTemp[[self$params$predictedCol]])[1], 0, 1)
       }
-
+      
       # Creating the groups for the grouped Lasso model.
       # Factor variables have a group that is one less than the number of levels.
       # Everything else has length one.
       # The length of this vector should be the same as the number of columns in modMat.
-      private$group <- rep(1:(ncol(private$dfTrainTemp) - 1 ),
-                  times <- sapply(private$dfTrainTemp[ ,!(colnames(private$dfTrainTemp) == self$params$predictedCol)],
-                               function(x) ifelse(is.factor(x), length(levels(x)) - 1, 1)))
-
+      private$group <-
+        rep(1:(ncol(private$dfTrainTemp) - 1),
+            times <-
+              sapply(private$dfTrainTemp[, !(colnames(private$dfTrainTemp) == self$params$predictedCol)],
+                     function(x)
+                       ifelse(is.factor(x), length(levels(x)) - 1, 1)))
+      
       if (length(private$group) != ncol(private$modMat)) {
         stop('There is a mismatch in group definition and model matrix definition')
         # This message should likely be refined, perhaps something different for
         # greater than or less than ...
       }
-
+      
       # Generate fit grLasso object
       familyModuleName <- ""
       if (self$params$type == 'classification') {
         familyModuleName <- "binomial"
       } else if (self$params$type == 'regression') {
         familyModuleName <- "gaussian"
       }
-
-      private$fitGrLasso <- cv.grpreg(X = private$modMat,
-                                      y = private$dfTrainTemp[[self$params$predictedCol]],
-                                      group = private$group,
-                                      #lambda = can enter values here,
-                                      #  but we will use default
-                                      family = familyModuleName,
-                                      penalty = "grLasso",
-                                      nfolds = 5)
+      
+      private$fitGrLasso <-
+        cv.grpreg(
```

Comment:
Do you mind adding `grpreg::` to line 319? While we're in these files, let's try to make sure each function call has ::